Assets/SplineMesh/Scripts/Editor/SplineEditor.cs

```diff
@@ -200,42 +253,54 @@ private enum SelectionType {
                 EditorUtility.SetDirty(target);
         }
 
-        bool Button(Vector2 position, GUIStyle style) {
+        bool Button(Vector2 position, GUIStyle style)
+        {
             return GUI.Button(new Rect(position - new Vector2(QUAD_SIZE / 2, QUAD_SIZE / 2), new Vector2(QUAD_SIZE, QUAD_SIZE)), GUIContent.none, style);
         }
 
-        public override void OnInspectorGUI() {
+        public override void OnInspectorGUI()
+        {
             serializedObject.Update();
             // hint
             EditorGUILayout.HelpBox("Hold Alt and drag a node to create a new one.", MessageType.Info);
 
-            if(spline.nodes.IndexOf(selection) < 0) {
+            if (spline.nodes.IndexOf(selection) < 0)
+            {
                 selection = null;
             }
 
             // add button
-            if (selection == null) {
+            if (selection == null)
+            {
                 GUI.enabled = false;
             }
-            if (GUILayout.Button("Add node after selected")) {
+            if (GUILayout.Button("Add node after selected"))
+            {
                 Undo.RecordObject(spline, "add spline node");
-                SplineNode newNode = new SplineNode(selection.Direction, selection.Direction + selection.Direction - selection.Position);
-                var index = spline.nodes.IndexOf(selection);
-                if(index == spline.nodes.Count - 1) {
-                    spline.AddNode(newNode);
-                } else {
-                    spline.InsertNode(index + 1, newNode);
-                }
-                selection = newNode;
-                serializedObject.Update();
+                spline.SplitAtTime(5.25f);
+
+                //SplineNode newNode = new SplineNode(selection.Direction, selection.Direction + selection.Direction - selection.Position);
+                //var index = spline.nodes.IndexOf(selection);
+                //if (index == spline.nodes.Count - 1)
+                //{
+                //    spline.AddNode(newNode);
+                //}
+                //else
+                //{
+                //    spline.InsertNode(index + 1, newNode);
+                //}
+                //selection = newNode;
+                //serializedObject.Update();
```

Comment:
this was a wrong commit, the code is restored in 4f6b3b1fd3fb84facfe68e812e79ed7acf326412